Prox/Prox/FirebasePlacesDatabase.swift

```diff
@@ -26,17 +26,14 @@ class FirebasePlacesDatabase: PlacesDatabase {
         geofire = GeoFire(firebaseRef: rootRef.child(GEOFIRE_PATH))
     }
 
-    // todo: handle errors (double-check every query)
-    // TODO: handle version in DB
     /*
      * Queries GeoFire to get the place keys around the given location and then queries Firebase to
      * get the place details for the place keys.
      */
     func getPlaces(forLocation location: CLLocation) -> Future<[DatabaseResult<Place>]> {
-        // TODO: OperationQueue.current okay?
-        let places = getPlaceKeys(aroundPoint: location).andThen(upon: OperationQueue.current!) { (placeKeyToLoc) -> Future<[DatabaseResult<Place>]> in
+        let queue = DispatchQueue.global(qos: .userInitiated)
```

Comment:
it's really interesting seeing you coming straight to swift 3.0 / iOS10 without being encumbered with the knowledge of how things used to work in swift 2.x / previous iOS versions. I had no idea about the `qos` option on `DispatchQueue` :)